.github/workflows/integration.yml

```diff
@@ -214,3 +196,52 @@ jobs:
 
     - name: Run frontend tests
       run: ./gradlew frontendTest -Pci
+
+  preview:
```

Comment:
How is this change related to ubuntu integration testing? I can see that you have added `runs-on` which wasn't there before; but you also seem to have modified parts of code.

Reply:
It is not related to Ubuntu specifically, but should only be run once. Hence, I extracted it out to a different job that will run after the integration tests have been completed. Code-wise nothing has been changed, only the `if` statements have been consolidated.

Reply:
I will prefer to see this change in a new PR or mention this in the commit message. I agree it's a small tweak but PRs should always just perform one specific change. This helps in keeping track and finding PRs and commits if ever required.

Reply:
@Tejas2805 I have reverted the PR back to e4e832d, as the original design did not accompany this change.